src/objectid.ts

```diff
@@ -170,34 +245,38 @@ export class ObjectId extends BSONValue {
    * Generate a 12 byte id buffer used in ObjectId's
    *
    * @param time - pass in a second based timestamp.
+   * @param buffer - Optionally pass in a buffer instance.
+   * @param offset - Optionally pass in a buffer offset.
    */
-  static generate(time?: number): Uint8Array {
+  static generate(time?: number, buffer?: Uint8Array, offset: number = 0): Uint8Array {
```

Comment:
Sorry I did not catch this the first time, we just have to give this the same treatment we did the new constructor. The overload that accepts the new parameters should be internal and the public one can remain the `time?` argument only.